sdk_container/src/third_party/coreos-overlay/profiles/coreos/base/profile.bashrc.slsa-provenance

```diff
@@ -1,194 +1,269 @@
+__slsa_rd_printf() {
+    local uri kind csum leading_comma
+    uri=${1}; shift
+    kind=${1}; shift
+    csum=${1}; shift
+    leading_comma=${1-x}
 
-    einfo "    Provenance: recording subject (output)"
+    printf '%s{ "uri": "%s", "digest": { "%s": "%s" } }\n' \
+           "${leading_comma:+,}" "${uri}" "${kind}" "${csum}"
+}
+# --
 
-    echo '   "subject": ['
+# Generate SLSA provenance 1.0 Subject information.
+# The information will cover all installation files shipped with a package.
+__slsa_provenance_subject_members() {
+    local parallel do_comma checksum filepath filepath_d
 
-    (
-        cd "$D"
-        find . -type f -print | sed 's:^./::' | xargs -P "$parallel" -L 1 sha512sum | sort -k2
-    ) | while read checksum filepath; do
-        echo -en "${comma}     {\"name\":\"/$filepath\", \"digest\":{\"sha512\":\"$checksum\"}}"
-        if [ -z "$comma" ] ; then
-            comma=',\n'
-        fi
-    done
-   echo -en "\n   ]"
+    einfo '    Provenance: recording subject (output)'
+
+    parallel=$(nproc)
+    do_comma=''
+    find "${D}" -type f -print0 | \
+        xargs -0 -P "${parallel}" -L 1 sha512sum | \
+        sort -k2 | \
+        while read -r checksum filepath; do
+            filepath_d=${filepath#"${D}/"}
+            __slsa_rd_printf "${filepath_d}" 'sha512' "${checksum}" "${do_comma}"
+            if [[ -z ${do_comma} ]] ; then
+                do_comma=x
+            fi
+        done
 }
 # --
 
-__slsa_provenance_materials() {
-    local csum="" uri="" repo="" ebuild="" ebuildcsum=""
-
-    local ebuild="${CATEGORY}/${PN}/${PF}.ebuild"
-    local repopath="$(portageq get_repo_path ${ROOT:-/} coreos)"
-    if [ -f "${repopath}/${ebuild}" ] ; then
-        repo="coreos-overlay"
-        ebuildcsum=$(sha1sum - < "${repopath}/${ebuild}")
-    else
-        repopath="$(portageq get_repo_path ${ROOT:-/} portage-stable)"
-        if [ -f "${repopath}/${ebuild}" ] ; then
-            repo="portage-stable"
-            ebuildcsum=$(sha1sum - < "${repopath}/${ebuild}")
+__slsa_current_repo() {
+    local ebuild=${1}; shift
+    local -n repo_ref=${1}; shift
+    local -n ebuild_full_path_ref=${1}; shift
+
+    local some_root sr_set v
+    sr_set=
+    for v in SYSROOT ROOT BROOT; do
+        if [[ -n ${!v:-} ]]; then
+            some_root=${!v%/}
+            # strip all trailing slashes, could be easier with extglob, but
+            # this is not guaranteed by PMS.
+            while [[ ${some_root%/} != "${some_root}" ]]; do
+                some_root=${some_root%/}
+            done
+            sr_set=x
+            break
         fi
+    done
+    if [[ -z ${sr_set} ]]; then
+        die "SLSA provenance: No root directory for portage configuration could be found"
     fi
-    if [ -z "${repo}" ]; then
-        die "SLSA provenance: Unable to detect ebuild repository for package '${ebuild}'"
+
+    local repos_conf
+    local -a locations
+    repos_conf="${some_root}/etc/portage/repos.conf"
```

Comment:
shouldn't this look at `PORTAGE_CONFIG_ROOT`?

Reply:
PORTAGE_CONFIGROOT is superseded by SYSROOT now. It's at least what `man emerge` tells me. Also, I don't remember if PORTAGE_CONFIGROOT was even available in the environment at this point.

Reply:
ok